torch/onnx/verification.py

```diff
@@ -243,6 +247,238 @@ def compare_ort_pytorch_model_with_input(input_args, input_kwargs):
+    opset_version: Optional[int] = None
+    do_constant_folding: bool = True
+    dynamic_axes: Optional[Mapping[str, Union[Mapping[int, str], Sequence[int]]]] = None
+    keep_initializers_as_inputs: Optional[bool] = None
+    custom_opsets: Optional[Mapping[str, int]] = None
+    export_modules_as_functions: Union[bool, Set[Type[torch.nn.Module]]] = False
+
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+
+class _GraphDiff:
+    """A class to represent the difference between two graphs."""
+
+    def __init__(self, graph_a: _C.Graph, graph_b: _C.Graph):
+        """Construct a _GraphDiff object.
+
+        Args:
+            graph_a (_C.Graph): First graph to compare.
+            graph_b (_C.Graph): Second graph to compare.
+        """
+        self.graph_a = graph_a
+        self.graph_b = graph_b
+
+    def __str__(self):
+        """See function :func:`_graph_diff`."""
+        return self.diff_report()
+
+    def _indent(self, s):
+        return "\n".join(["\t" + line for line in s.splitlines()])
+
+    def diff_report(self):
+        """Return a string representation of the graph difference.
+
+        The report shows the first pair of nodes that diverges. It also shows the source
+        location of the pair of nodes.
+
+        Returns:
+            graph_diff_report (str): A string representation of the graph difference.
+        """
+        graph_a = self.graph_a
+        graph_b = self.graph_b
+
+        graph_a_str = str(graph_a)
+        graph_b_str = str(graph_b)
+
+        if graph_a_str == graph_b_str:
+            return ""
+
+        graph_diff = difflib.ndiff(
+            graph_a_str.splitlines(True), graph_b_str.splitlines(True)
+        )
+        graph_diff_report = "Graph diff:\n" + self._indent("".join(graph_diff)) + "\n"
+
+        for n_ref, n_check in zip(graph_a.nodes(), graph_b.nodes()):
+            if str(n_ref) != str(n_check):
+                graph_diff_report += "First diverging operator:\n"
+                node_diff = difflib.ndiff(
+                    str(n_ref).splitlines(True), str(n_check).splitlines(True)
+                )
+                source_printout = f"node diff:\n{self._indent(''.join(node_diff))}\n"
+
+                ref_stack = n_ref.sourceRange()
+                if ref_stack:
+                    source_printout += (
+                        f"Reference source location:\n{self._indent(ref_stack)}\n"
+                    )
+                check_stack = n_check.sourceRange()
+                if check_stack:
+                    source_printout += (
+                        f"Check source location:\n{self._indent(check_stack)}\n"
+                    )
+
+                graph_diff_report += source_printout
+
+                break
+
+        return graph_diff_report
+
+
+def _check_jit_model_diff(
+    model: Union[torch.nn.Module, torch.jit.ScriptModule],
+    test_inputs: Sequence[Tuple[Tuple[Any, ...], Mapping[str, Any]]],
```

Comment:
This param may need some explanation on what is expected

Reply:
adjusted naming and docstring.

Reply:
adjusted naming and docstring.